cg/services/order_validation_service/workflows/tomte/validation_service.py

```diff
@@ -25,15 +29,14 @@
 
 class TomteValidationService(OrderValidationService):
 
-    def __init__(self, field_validator: TomteModelValidator, store: Store):
-        self.field_validator = field_validator
+    def __init__(self, store: Store):
         self.store = store
 
     def validate(self, order_json: str) -> ValidationErrors:
-        order, field_errors = self.field_validator.validate(order_json)
-
-        if field_errors:
-            return field_errors
+        try:
+            order: TomteOrder = TomteModelValidator.validate(order_json)
+        except ValidationError as error:
+            return convert_errors(error.errors())
```

Comment:
How will this work, will all validation errors be contained in the ValidationError? Also, would it not be nice to encapsulate the logic for mapping the errors from pydantic to the general error structure in the actual field validator?

Reply:
Might be! I do think they should be raised in that case though! The return type was messy

Reply:
Info on the .errors() https://docs.pydantic.dev/latest/api/pydantic_core/#pydantic_core.ValidationError.errors

Reply:
The general error object we have created is not written as an exception, so not sure that works. We can do the remapping here and scrap the field validator class 👍

Reply:
Is the `convert_errors` method not hiding it already? Or we should not need to know that the errors are converted?

Reply:
It is! I guess it is unlikely we will need more than a function call to do it